File: coding-practice/LeetCode 75/Easy/14 - Longest Common Prefix/Python/Solution.py
strs = ["flower","flow","flight"]

class Solution:
    def __init__(self) -> None:
        pass

    def longestCommonPrefix(self, listOfStrings):
        # If the list string is empty
        if not listOfStrings:       
            return ""
        
        prefix = listOfStrings[0]
        for string in listOfStrings[1:]:
            # condition if current string does not begins with prefix
            while string[:len(prefix)] != prefix:
                # gradually remove the last element on the prefix
                prefix = prefix[:-1]
                # if prefix eventually becomes empty, then it means there are no common prefix in listOfString
                if not prefix:
                    return ""

        # return prefix
        return prefix
    
    def display_solution(self):
        loop = True

        while loop:
            print("Longest Common Prefix")
            print("This function will search for the longest common prefix of a list of strings")
            input_string = input("Enter a list of strings separated by commas: ")
            words = input_string.split(",")
            longestPrefix = self.longestCommonPrefix(words)
            print(f'Longest common prefix of {words} is \'{longestPrefix}\'')
            while True:
                print("Would you like to repeat? (y/n)")
                response = input()
                response = "" if response is None else response

                if response == "":
                    print("Please choose the option")
                elif response == "y" or response == "Y":
                    loop = True
                    break
                elif response == "n" or response == "N":
                    loop = False
                    break
                else:
                    print("Invalid input. Please enter 'y' or 'n'")

    
obj = Solution()
obj.display_solution()

File: coding-practice/LeetCode 75/Easy/14 - Longest Common Prefix/C++/Solution.cpp
#include<iostream>
#include<cmath>
#include<vector>
#include<sstream>

using namespace std;

class Solution{
	private:
	
	
	public:
		void HelloWorld(){
			cout << "Hello World";	
		};
		
		string longestCommonPrefix(vector<string>& words){
			if(words.size() == 0){
				return "";
			}
			
			string prefix = words[0];
			
			for(int i = 1; i < words.size(); i++){
				while(words[i].substr(0, prefix.size()) != prefix){
					prefix.resize(prefix.size() - 1);
					if(prefix.empty()){
						return "";
					}
				}
			}
			
			return prefix;	
		};
		
		void
};

vector<string> split(const string&, char delimiter){
	vector<string> tokens;
	string token;
	istringStream tokenStream()
};

int main(){
	/*
	Solution obj;
	string arr_words[] = {"flower","flow","flight"};
	vector<string> words(arr_words, arr_words + sizeof(arr_words) / sizeof(string));
	string longestPrefix = obj.longestCommonPrefix(words);
	
	cout << "The longest common prefx of {";
	for(int i = 0; i < words.size(); i++){
		if(i + 1 != words.size()){
			cout << words[i] << ", ";
		}
		else{
			cout << words[i];
		}
	}
	cout << "} is " << longestPrefix << endl;
	*/
	
	string input_string = "apple,banana,cherry";
	char delimiter = ',';
	vector<string> result = 
	return 0;
}

